messaging/messaging_utils.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package messaging
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	bareTopicNamePattern = regexp.MustCompile("^[a-zA-Z0-9-_.~%]+$")
+	colorPattern         = regexp.MustCompile("^#[0-9a-fA-F]{6}$")
+)
+
+func validateMessage(message *Message) error {
+	if message == nil {
+		return fmt.Errorf("message must not be nil")
+	}
+
+	target := bool2int(message.Token != "") + bool2int(message.Condition != "") + bool2int(message.Topic != "")
```

Comment:
I think that 
```
target := 0
for _, v := range []bool{message.Token != "", message.Condition != "", message.Topic != ""} {
  if v {
    target +=1
}
```

is slightly more readable, I don't feel very strongly about this though. 

maybe make a function of it at a slightly higher level than the single translation

```
func countTrues(arr []bool) {
  res := 0 
    for _, v := range arr {
      if v {
        res += 1
      }
  }
 return res
}
```
or better yet
```
func countNonEmpty(arr []string) {
  res := 0 
    for _, s := range arr {
      if s != "" {
        res += 1
      }
  }
 return res
}

Reply:
Good idea. Done